### PR TITLE
tests: fs: fat_fs_api: Add support for non-"flash map" devices

### LINKED ISSUE #53151: fs: FAT_FS_API MKFS test uses driver specific calls
**Describe the bug**
The FAT_FS_API MKFS test uses flash-specific apis in the test, causing build errors when the test is built for a non-flash device using disk access to interact with a file system

**Expected behavior**
I expected the test to be more portable and be usable to test any device using a fat fs filesystem, not just flash, and also I expected not to get build errors

**Impact**
It was blocking #50019 due to CI but I have proposed a hotfix PR to hopefully work around it

### PR DESCRIPTION
**Pull request by de-nordic**
Test will now use disk_access_ functions to erase FAT FS disk before some operations when target disk is not set with CONFIG_DISK_DRIVER_FLASH.

Fixes #53151

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

### CODE REVIEW

**Comment by carlescufi** [comment]
@decsny Could you please review this?

File: tests/subsys/fs/fat_fs_api/src/test_fat.h [hunk 1]
```diff
@@ -9,7 +9,17 @@
...
+#define DISK_NAME CONFIG_DISK_RAM_VOLUME_NAME
+#elif defined(CONFIG_DISK_DRIVER_FLASH)
+#define DISK_NAME DT_PROP(DT_NODELABEL(test_disk), disk_name)
+#elif defined(CONFIG_DISK_DRIVER_SDMMC)
+#define DISK_NAME SDMMC_VOLUME_NAME
+#else
+#error "Failed to select DISK access type"
+#endif
+
+#define FATFS_MNTP	"/"DISK_NAME":"
```
Comment (nordicjm) [inline tests/subsys/fs/fat_fs_api/src/test_fat.h:22]: For lfs-based tests and samples (including smp_svr) mount points are just using /<name/ without a colon e.g. /lfs1/ or /lfs3/ - this is the same style as device numbering for e.g. i2c and spi: i2c0, i2c1, spi3, spi82, etc. So with that, do we want to have FAT be the odd one out?

**Review by nordicjm — DISMISSED** [review]

File: tests/subsys/fs/fat_fs_api/src/test_fat.h (hunk 1 above)
Comment (de-nordic) [reply tests/subsys/fs/fat_fs_api/src/test_fat.h:22]: The FAT has not the best way of figuring out paths and the ":" is required unless me modify FAT zephyrproject-rtos/fatfs, which is slightly modified (just to make it work with us) ELM FAT driver.

File: tests/subsys/fs/fat_fs_api/src/test_fat.h [hunk 2]
```diff
@@ -9,7 +9,17 @@
...
 #include <zephyr/fs/fs.h>
 
-#define FATFS_MNTP	"/NAND:"
+#ifdef CONFIG_DISK_DRIVER_RAM
+#define DISK_NAME CONFIG_DISK_RAM_VOLUME_NAME
+#elif defined(CONFIG_DISK_DRIVER_FLASH)
+#define DISK_NAME DT_PROP(DT_NODELABEL(test_disk), disk_name)
+#elif defined(CONFIG_DISK_DRIVER_SDMMC)
+#define DISK_NAME SDMMC_VOLUME_NAME
+#else
```
Comment (decsny) [inline tests/subsys/fs/fat_fs_api/src/test_fat.h:18]: please add support for MMC in this PR (relevant kconfigs here: CONFIG_DISK_DRIVER_MMC and MMC_VOLUME_NAME)

**Review by decsny — CHANGES_REQUESTED** [review]

File: tests/subsys/fs/fat_fs_api/src/test_fat.h (hunk 2 above)
Comment (de-nordic) [reply tests/subsys/fs/fat_fs_api/src/test_fat.h:20]: Done.
Also fixed the line 17, by adding missing `CONFIG_`

**Review by decsny — COMMENTED** [review]
hold tight I am going to make a fix for the CI issues you have, there is an issue with MMC

**Comment by decsny** [comment]
> hold tight I am going to make a fix for the CI issues you have, there is an issue with MMC

#54571 should help this PR

**Review by nordicjm — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**